rewrite-pattern.js

```diff
@@ -728,9 +803,42 @@ const rewritePattern = (pattern, flags, options) => {
 	};
 
 	const tree = parse(pattern, flags, regjsparserFeatures);
+
+	if (regjsparserFeatures.modifiers) {
+		const allDisabledModifiers = Object.create(null)
+		const itemStack = [tree];
+		let node;
+		while (node = itemStack.pop(), node != undefined) {
+			if (Array.isArray(node)) {
+				Array.prototype.push.apply(itemStack, node);
+			} else if (typeof node == 'object' && node != null) {
+				for (const key of Object.keys(node)) {
+					const value = node[key];
+					if (key == "modifierFlags") {
+						if (value.disabling.length > 0){
+							value.disabling.split("").forEach((flag)=>{
+								allDisabledModifiers[flag] = true
+							});
+						}
+					} else if (typeof value == 'object' && value != null) {
+						itemStack.push(value);
+					}
+				}
+			}
+		}
+
+		for (const flag of Object.keys(allDisabledModifiers)) {
+			config.modifiersData[flag] = true;
```

Comment:
Why do we need to look for modifiers that are disabled in the regexp to populate this object? Can we just look at which `flags` are passed to the regexp?

Reply:
Because I want to remove `i` only for the whole expression where `-i` exists. For example `/a(((((((?-i:a)))))))/i`, at this time we need to know that `-i` exists when processing the first a.